packages/fast-usdc/test/fast-usdc.contract.test.ts

```diff
@@ -510,115 +542,460 @@ test('LP repay', async t => {
+        numerator: add(par250.numerator, split.PoolFee),
+      },
+      totalBorrows: { value: 105839999n },
+      totalContractFees: { value: 432000n },
+      totalPoolFees: { value: 1728001n },
+      totalRepays: { value: 105839999n },
+    },
+    'metrics after advancing',
+  );
+});
+
+test.todo(
+  'PERF: Target: settlement completes in a few minutes (after USDC is minted)',
+);
+
+// most likely in exo unit tests
+test.todo(
+  'C21 - Contract MUST log / timestamp each step in the transaction flow',
+);
+
+test.serial('STORY03: see accounting metrics', async t => {
+  const {
+    common: {
+      brands: { usdc },
+    },
+    startKit: { metricsSub },
+  } = t.context;
+  const { value: metrics } = await E(metricsSub).getUpdateSince();
+
+  t.log(metrics);
+  t.true(matches(metrics, PoolMetricsShape));
+});
+test.todo('document metrics storage schema');
+test.todo('get metrics from vstorage');
+
+test.serial('STORY05: LP collects fees on 100 USDC', async t => {
+  const {
+    sync,
+    common: {
+      brands: { usdc },
+    },
+  } = t.context;
+
+  const lp = await sync.lp.promise;
+  const got = await E(lp.lp200).withdraw(t, 0.5); // redeem 1/2 my shares
+
+  // C3 - Contract MUST calculate ...
+  // Mostly, see unit tests for calculateAdvance, calculateSplit
+  // TODO: add a feeTools unit test for the magic number below.
+  t.deepEqual(got, add(usdc.units(100), usdc.make(691_200n)));
+
+  await E(lp.lp200).deposit(t, 100_000_000n); // put all but the fees back in
+});
+
+test.serial('With 250 available, 3 race to get ~100', async t => {
+  const {
+    bridges: { snapshot, since },
+    evm: { cctp, txPub },
+    common: {
+      commonPrivateArgs: { feeConfig },
+      utils: { transmitTransferAck },
+    },
+    startKit: { metricsSub },
+    mint,
+  } = t.context;
+
+  const cust = {
+    racer1: makeCustomer('Racer1', cctp, txPub.publisher, feeConfig),
+    racer2: makeCustomer('Racer2', cctp, txPub.publisher, feeConfig),
+    racer3: makeCustomer('Racer3', cctp, txPub.publisher, feeConfig),
+  };
+
+  await cust.racer3.checkPoolAvailable(t, 125_000_000n, metricsSub);
+
+  const bridgePos = snapshot();
+  const [sent1, sent2, sent3] = await Promise.all([
+    cust.racer1.send(t, 110_000_000n, 'osmo1234a'),
+    cust.racer2.send(t, 120_000_000n, 'osmo1234b'),
+    cust.racer3.send(t, 125_000_000n, 'osmo1234c'),
+  ]);
+  cust.racer1.checkSent(t, since(bridgePos));
+  cust.racer2.checkSent(t, since(bridgePos));
+  // TODO/WIP: cust.racer3.checkSent(t, since(bridgePos), 'forward - LP depleted');
   await transmitTransferAck();
-  // Nothing we can check here, unless we want to inspect calls to `trace`.
-  // `test/exos/advancer.test.ts` covers calls to `log: LogFn` with mocks.
-  // This is still helpful to call, so we can observe "Advance transfer
-  // fulfilled" in the test output.
+  await transmitTransferAck();
+  await transmitTransferAck();
+  await Promise.all([mint(sent1), mint(sent2), mint(sent3)]);
```

Comment:
This is cool, but I'm struggling to understand how we plan to check the results of the Promise.all. I don't think we'll have any bridge messages that indicate insufficient funds (advancer's `log` would know though.) Will we wait for txs to be posted to vstorage?

Regardless, consider specifying in the TODO/WIP

Reply:
Yes, as I was working on this test, I realized that our mock bridge stuff only works one tx at a time. I'm not inclined to go on at any length about what's TODO/WIP here just now.